src/QMCDrivers/WFOpt/QMCOptimizeBatched.h

```diff
@@ -0,0 +1,116 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2020 QMCPACK developers.
+//
+// File developed by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
+//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//                    Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+/** @file QMCOptimizeBatched.h
+ * @brief Definition of QMCDriver which performs VMC (using batched driver) and optimization.
+ */
+#ifndef QMCPLUSPLUS_QMCOPTIMIZATION_VMCBATCHED_H
+#define QMCPLUSPLUS_QMCOPTIMIZATION_VMCBATCHED_H
+
+#include <memory>
+#include "QMCDrivers/QMCDriver.h"
+#include "Optimize/OptimizeBase.h"
+#include "QMCDrivers/QMCDriverNew.h"
+#include "QMCDrivers/QMCDriverInput.h"
+#include "QMCDrivers/VMC/VMCDriverInput.h"
+#include "QMCDrivers/VMC/VMCBatched.h"
+
+
+namespace qmcplusplus
+{
+///forward declaration of a cost function
+class QMCCostFunctionBase;
+class HamiltonianPool;
+class MCPopulation;
+
+/** @ingroup QMCDrivers
+ * @brief Implements wave-function optimization
+ *
+ * Optimization by correlated sampling method with configurations
+ * generated from VMC.
+ */
+
+class QMCOptimizeBatched : public QMCDriver
```

Comment:
QMCDriver will not be retained after the shift to unified drivers is done.  This should be refactored to not rely on inheriting from it. Considerable confusion will definitely result.